### PR TITLE
Dont't break DefaultImageScalingFactory, if for any reason the fieldn…

### PR DESCRIPTION
**Pull request by thet**
…ame isn't available on the context.

### CODE REVIEW

File: plone/namedfile/scaling.py [hunk 1]
```diff
@@ -192,18 +193,18 @@ def create_scale(self, data, direction, height, width, **parameters):
...
 
     def __call__(
-        self,
-        fieldname=None,
-        direction='thumbnail',
-        height=None,
-        width=None,
-        scale=None,
-        **parameters
+            self,
```
Comment (loechel) [inline plone/namedfile/scaling.py]: why is there another level of indention?

File: plone/namedfile/scaling.py [hunk 2]
```diff
@@ -66,15 +67,15 @@ def absolute_url(self):
         return self.url
 
     def srcset_attribute(self):
-        srcset_attr = []
+        _srcset_attr = []
```
Comment (loechel) [inline plone/namedfile/scaling.py]: why is this rename necessary?

File: plone/namedfile/scaling.py [hunk 3]
```diff
@@ -66,15 +67,15 @@ def absolute_url(self):
...
         extension = self.data.contentType.split('/')[-1].lower()
         for scale in self.srcset:
-            srcset_attr.append(u'{0}/@@images/{1}.{2} {3}x'.format(
+            _srcset_attr.append(u'{0}/@@images/{1}.{2} {3}x'.format(
                 self.context.absolute_url(),
                 scale['uid'],
                 extension,
                 scale['scale']))
-        srcset_attr = ', '.join(srcset_attr)
+        srcset_attr = ', '.join(_srcset_attr)
```
Comment (loechel) [inline plone/namedfile/scaling.py]: wouldn't be a direct return shorter and the same?

File: plone/namedfile/scaling.py [hunk 4]
```diff
@@ -192,18 +193,18 @@ def create_scale(self, data, direction, height, width, **parameters):
...
+            height=None,
+            width=None,
+            scale=None,
+            **parameters
     ):
 
         """Factory for image scales`.
         """
-        orig_value = getattr(self.context, fieldname)
+        orig_value = getattr(self.context, fieldname, None)
```
Comment (loechel) [inline plone/namedfile/scaling.py]: this is redundant, or only explicit. getattr will always return None if the attribute could not be found.

Comment (thet) [inline plone/namedfile/scaling.py]: No, getattr raises an AttributeError, if the attribute cannot be found:
```
>>> getattr(None, 'bla')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'bla'
```

File: plone/namedfile/scaling.py (hunk 1 above)
Comment (thet) [inline plone/namedfile/scaling.py]: Actually, just because my linter said so.
The reason for this formatting is, that this way it isn't on the same level like the inner function block code.
Although that's also just a question of taste.

File: plone/namedfile/scaling.py (hunk 3 above)
Comment (thet) [inline plone/namedfile/scaling.py]: True. Again, my linter complained, and I think this formatting rule is ok. The linter said, here the type of the variable is changed from a list to a string.
I don't have a strong opinion on this. Should I change it back?

File: plone/namedfile/scaling.py (hunk 4 above)
Comment (loechel) [inline plone/namedfile/scaling.py]: Ok, I did expect the other way.

**Comment by loechel** [comment]
as long as code-analysis did not complain about plone coding conventions I am fine.

File: plone/namedfile/scaling.py (hunk 4 above)
Comment (thet) [inline plone/namedfile/scaling.py]: Actually, that context doesn't have the fieldname attribute should not happen in the first place. But I encountered an error at a project which was caused by this...

**Comment by thet** [comment]
@loechel tnx for the review.
